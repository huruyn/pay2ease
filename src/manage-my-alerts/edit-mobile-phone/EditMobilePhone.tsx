import {
  Col, Modal, Row, Form,
} from 'react-bootstrap';

export interface EditMobilePhoneContentProps {
  handleCancel: () => void;
  handleNext: () => void;
}

export const EditMobilePhoneContent = ({
  handleCancel,
  handleNext,
}: EditMobilePhoneContentProps) => (
  <>
    <Row style={{ padding: '30px' }}>
      <h3>MOBILE PHONE NUMBER UPDATE</h3>
      <ul style={{ marginLeft: '25px' }}>
        <li style={{ marginBottom: '6px' }}>
          This will be your SMS (Short Message Service) mobile phone number on
          file with ZippyX.
        </li>
        <li style={{ marginBottom: '6px' }}>
          If you choose to recieve SMS alerts, ZippyX will use this number to
          send alerts to you.
        </li>
        <li style={{ marginBottom: '6px' }}>
          The alerts service is only available to Canadian wireless device
          numbers.
        </li>
        <li style={{ marginBottom: '6px' }}>
          Standard messaging and data rates may apply, ZippyX will send you a
          monthly reminder through SMS text message to allow you to opt out of
          recieving SMS text message from us.
        </li>
      </ul>
      <Row
        style={{
          display: 'flex',
          justifyContent: 'space-evenly',
          alignItems: 'center',
          margin: '30px 0px',
        }}
      >
        <Col>
          <Form.Group
            as={Row}
            className="mb-3"
            controlId="MobilePhone"
            style={{ display: 'flex', alignItems: 'center' }}
          >
            <Col md={4}>
              <Form.Label column>Mobile phone number:</Form.Label>
            </Col>
            <Col md={8}>
              <Form.Control type="tel" style={{ width: '120px' }} required />
            </Col>
          </Form.Group>
        </Col>
      </Row>
    </Row>
    <Row
      style={{
        display: 'flex',
        justifyContent: 'space-between',
        marginTop: '10px',
        padding: '20px 10px',
        backgroundColor: '#e3e3e3',
        margin: '0px',
      }}
    >
      <Col>
        <button
          type="button"
          id="cancel-btn-edit-mobile-phone"
          style={{
            backgroundColor: 'transparent',
            color: 'black',
            padding: '5px 12px',
            border: '0',
          }}
          onClick={handleCancel}
        >
          <i />
          Cancel
        </button>
      </Col>
      <Col style={{ textAlign: 'end' }}>
        <button
          type="button"
          id="next-btn-edit-mobile-phone"
          style={{
            borderRadius: '3px',
            padding: '5px 12px',
            backgroundColor: '#C41F3E',
            borderColor: '#C41F3E',
            color: 'white',
          }}
          onClick={handleNext}
        >
          Next
        </button>
      </Col>
    </Row>
  </>
);
export interface EditMobilePhoneProps extends EditMobilePhoneContentProps {
  show: boolean;
}
const EditMobilePhone = ({
  show,
  handleCancel,
  handleNext,
}: EditMobilePhoneProps) => (
  <Modal animation={false} show={show} size="lg">
    <Modal.Body style={{ padding: '0px' }}>
      <EditMobilePhoneContent
        handleCancel={handleCancel}
        handleNext={handleNext}
      />
    </Modal.Body>
  </Modal>
);

export default EditMobilePhone;
