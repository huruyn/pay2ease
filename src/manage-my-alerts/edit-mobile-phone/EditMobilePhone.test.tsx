import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { EditMobilePhoneContent } from './EditMobilePhone';

const ReactTestRenderer = require('react-test-renderer');

describe('EditMobilePhone Component', () => {
  it('matches the snapshot', () => {
    const handleCancel = jest.fn();
    const handleNext = jest.fn();
    const tree = ReactTestRenderer.create(
      <EditMobilePhoneContent handleNext={handleNext} handleCancel={handleCancel} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Click cancel button on EditMobilePhone', () => {
    const handleCancel = jest.fn();
    const handleNext = jest.fn();
    const { container } = render(
      <EditMobilePhoneContent handleNext={handleNext} handleCancel={handleCancel} />,
    );
    const mEvent = { preventDefault: jest.fn() };
    const btnElements = container.querySelectorAll('#cancel-btn-edit-mobile-phone');
    expect(btnElements).toHaveLength(1);
    fireEvent.click(btnElements[0], mEvent);
    expect(handleCancel).toBeCalledTimes(1);
  });
  it('Click save button on EditMobilePhone', () => {
    const handleClose = jest.fn();
    const handleNext = jest.fn();
    const { container } = render(
      <EditMobilePhoneContent handleNext={handleNext} handleCancel={handleClose} />,
    );
    const mEvent = { preventDefault: jest.fn() };
    const btnElements = container.querySelectorAll('#next-btn-edit-mobile-phone');
    expect(btnElements).toHaveLength(1);
    fireEvent.click(btnElements[0], mEvent);
    expect(handleNext).toBeCalledTimes(1);
  });
});
