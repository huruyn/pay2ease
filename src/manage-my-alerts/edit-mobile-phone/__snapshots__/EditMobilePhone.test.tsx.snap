// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMobilePhone Component matches the snapshot 1`] = `
Array [
  <div
    className="row"
    style={
      Object {
        "padding": "30px",
      }
    }
  >
    <h3>
      MOBILE PHONE NUMBER UPDATE
    </h3>
    <ul
      style={
        Object {
          "marginLeft": "25px",
        }
      }
    >
      <li
        style={
          Object {
            "marginBottom": "6px",
          }
        }
      >
        This will be your SMS (Short Message Service) mobile phone number on file with ZippyX.
      </li>
      <li
        style={
          Object {
            "marginBottom": "6px",
          }
        }
      >
        If you choose to recieve SMS alerts, ZippyX will use this number to send alerts to you.
      </li>
      <li
        style={
          Object {
            "marginBottom": "6px",
          }
        }
      >
        The alerts service is only available to Canadian wireless device numbers.
      </li>
      <li
        style={
          Object {
            "marginBottom": "6px",
          }
        }
      >
        Standard messaging and data rates may apply, ZippyX will send you a monthly reminder through SMS text message to allow you to opt out of recieving SMS text message from us.
      </li>
    </ul>
    <div
      className="row"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-evenly",
          "margin": "30px 0px",
        }
      }
    >
      <div
        className="col"
      >
        <div
          className="mb-3 row"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <div
            className="col-md-4"
          >
            <label
              className="form-label col-form-label col"
              htmlFor="MobilePhone"
            >
              Mobile phone number:
            </label>
          </div>
          <div
            className="col-md-8"
          >
            <input
              className="form-control"
              id="MobilePhone"
              required={true}
              style={
                Object {
                  "width": "120px",
                }
              }
              type="tel"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
    style={
      Object {
        "backgroundColor": "#e3e3e3",
        "display": "flex",
        "justifyContent": "space-between",
        "margin": "0px",
        "marginTop": "10px",
        "padding": "20px 10px",
      }
    }
  >
    <div
      className="col"
    >
      <button
        id="cancel-btn-edit-mobile-phone"
        onClick={[MockFunction]}
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "0",
            "color": "black",
            "padding": "5px 12px",
          }
        }
        type="button"
      >
        <i />
        Cancel
      </button>
    </div>
    <div
      className="col"
      style={
        Object {
          "textAlign": "end",
        }
      }
    >
      <button
        id="next-btn-edit-mobile-phone"
        onClick={[MockFunction]}
        style={
          Object {
            "backgroundColor": "#C41F3E",
            "borderColor": "#C41F3E",
            "borderRadius": "3px",
            "color": "white",
            "padding": "5px 12px",
          }
        }
        type="button"
      >
        Next
      </button>
    </div>
  </div>,
]
`;
